Gcpe.Hub.API.IntegrationTests/Views/TestSocialMediaViews.cs

```diff
@@ -0,0 +1,133 @@
+            }
+
+            var response = await _client.GetAsync("/api/socialmedia");
+            response.EnsureSuccessStatusCode();
+
+            var body = await response.Content.ReadAsStringAsync();
+            var deserializedBody = JsonConvert.DeserializeObject<SocialMediaPostViewModel[]>(body);
+
+            Assert.NotEmpty(deserializedBody);
+            deserializedBody.Should().HaveCountGreaterOrEqualTo(5);
+        }
+
+        [Fact]
+        public async Task Create_EndpointShouldReturnSuccessAndCorrectPost()
+        {
+            var stringContent = new StringContent(JsonConvert.SerializeObject(testPost), Encoding.UTF8, "application/json");
+
+            var response = await _client.PostAsync("/api/socialmedia", stringContent);
+            response.EnsureSuccessStatusCode();
+            var body = await response.Content.ReadAsStringAsync();
+            var messageResult = JsonConvert.DeserializeObject<SocialMediaPostViewModel>(body);
+
+            messageResult.Url.Should().Be(testPost.Url);
+        }
+
+        [Fact]
+        public async Task Create_EndpointShouldRequireUrl()
+        {
+            var brokenTestPost = testPost;
+            brokenTestPost.Url = null;
+            var stringContent = new StringContent(JsonConvert.SerializeObject(brokenTestPost), Encoding.UTF8, "application/json");
+
+            var response = await _client.PostAsync("/api/socialmedia", stringContent);
+            response.IsSuccessStatusCode.Should().BeFalse();
+        }
+
+        [Fact]
+        public async Task Get_EndpointShouldReturnSuccessAndCorrectPost()
+        {
+            var stringContent = new StringContent(JsonConvert.SerializeObject(testPost), Encoding.UTF8, "application/json");
+            var createResponse = await _client.PostAsync("/api/socialmedia", stringContent);
+            var createBody = await createResponse.Content.ReadAsStringAsync();
+            var createdPost = JsonConvert.DeserializeObject<SocialMediaPostViewModel>(createBody);
+            var id = createdPost.Id;
+
+            var response = await _client.GetAsync($"/api/socialmedia/{id}");
+            response.EnsureSuccessStatusCode();
+            var body = await response.Content.ReadAsStringAsync();
+            var postResult = JsonConvert.DeserializeObject<SocialMediaPostViewModel>(body);
+
+            postResult.Url.Should().Be(postResult.Url);
+            postResult.Id.Should().Be(id);
+        }
+
+        [Fact]
+        public async Task Get_EndpointShouldReturnNotFound()
+        {
+            var response = await _client.GetAsync($"/api/socialmedia/{Guid.NewGuid()}");
+
+            response.IsSuccessStatusCode.Should().BeFalse();
+        }
+
+        [Fact]
+        public async Task Put_EndpointShouldReturnSuccessAndCorrectMessage()
+        {
+            var stringContent = new StringContent(JsonConvert.SerializeObject(testPost), Encoding.UTF8, "application/json");

```

Comment:
There's lots of code duplication with the Get method. Can you either create a Theory or move the common code into a function?

Reply:
I think a bigger problem with these tests is that they are using multiple views in one test. If something breaks on the create endpoint multiple tests will fail instead of just the ones that test the create endpoint. I will instead save these posts directly using the DB context